src/mesh/nemesis_io_helper.C

```diff
@@ -2660,77 +2675,90 @@ Nemesis_IO_Helper::write_element_values(const MeshBase & mesh,
           const std::vector<dof_id_type> & elem_ids =
             (it == subdomain_map.end()) ? empty_vec : it->second;
 
-          // Possibly skip this (variable, subdomain) combination
-          if (active_subdomains.empty() || active_subdomains.count(sbd_id))
+          // Possibly skip this (variable, subdomain) combination. Also check that there is at least
+          // one element on the sbd... Apparently, it is possible to have a sbd with zero elements,
+          // e.g., when running "adaptivity_ex3" in parallel with the 'dimension=1' argument.
+          if ((active_subdomains.empty() || active_subdomains.count(sbd_id)) && elem_ids.size())
```

Comment:
elem_ids is only going to have local elements, right?  So some processors are going to be entering this block now while others skip it?  But that's okay because we're operating only on `current_local_solution` inside the block, not calling anything `parallel_object_only()`?

Reply:
I think so. `elem_ids` comes from the subdomain object, and we force it to be an empty vector if nothing is found:

```
for (const int sbd_id_int : global_elem_blk_ids)
  {
    const subdomain_id_type sbd_id =
      cast_int<subdomain_id_type>(sbd_id_int);
    auto it = subdomain_map.find(sbd_id);
    const std::vector<dof_id_type> empty_vec;
    const std::vector<dof_id_type> & elem_ids =
      (it == subdomain_map.end()) ? empty_vec : it->second;
```

So I think it is safe to just skip the whole routine if `elem_ids` is empty. That just means that there's no elements on the subdomain, which happens if there's more processors than elements in the global domain.